### PR TITLE
overrides: pyee>10.0.0 fixed

### PR DESCRIPTION
**Pull request by TyberiusPrime**
Pyee removed their setup.py (in a setuptools project), and the old patch thefore fails

Closes: #1587

- [x] Commit messages or code include _references to related issues or PRs_ (including third parties)
- [x] Commit messages are _[conventional](https://www.conventionalcommits.org/)_ - examples from [the log](https://github.com/nix-community/poetry2nix/commits/master) include "feat: add changelog files to fixup hook", "fix(contourpy): allow wheel usage", and "test: add sqlalchemy2 test"

### CODE REVIEW

File: overrides/default.nix [hunk 1]
```diff
@@ -2158,6 +2158,24 @@ lib.composeManyExtensions [
         ];
       });
 
+      polars = prev.polars.overridePythonAttrs (
+        old: {
+           cargoDeps = let
+             unpacked_src = pkgs.runCommand "polars-cargo-lock" {} ''
```
Comment (cpcloud) [inline overrides/default.nix:2164]: Does this use IFD? I can't recall what the exact criteria for that are.

File: overrides/default.nix [hunk 2]
```diff
@@ -3847,11 +3865,13 @@ lib.composeManyExtensions [
       # manually specify its version.
       pyee = prev.pyee.overrideAttrs (
         old: {
-          postPatch = old.postPatch or "" + ''
+          postPatch = old.postPatch or "" +
+          (if lib.versionOlder old.version "10.0.0" then
```
Comment (cpcloud) [inline overrides/default.nix:3869]: You can use `lib.optionalString` here for a bit more concision.

**Comment by TyberiusPrime** [comment]
I must have mixed up my git branches, the polar's stuff is not supposed to be in this pr. 
Sorry!.

I also think it's not IDF, since there's no nix code being imported. 

lib.optionalString is of course a good idea.

File: overrides/default.nix (hunk 2 above)
Comment (TyberiusPrime) [reply overrides/default.nix]: I can probably even skip the parenthesis, but I'm always wary about nix order of operations.

**Review by cpcloud — APPROVED** [review]

**Pull request merged**